src/requests/UsersApi.ts

```diff
@@ -0,0 +1,192 @@
+
+        return queryResult;
+    }
+
+}
+
+    /**
+     * This function is used to get the rank tag of a user. Returns the highest rank tag of the user. Else returns undefined.
+     * @param user The User or CurrentUser object to get the rank tag from.
+     * @returns The rank tag of the user. If the user has no rank tag, returns undefined.
+     */
+    export function getVRCRankTag(user: VRCAPI.Users.Models.User | VRCAPI.Users.Models.CurrentUser): VRCAPI.Users.Models.VRCRanks | undefined {
+        // the highest vrcrank we can find in the user's tag is the rank of the user we should return
+        const tags: VRCAPI.Generics.AllTags[] = user.tags;
+        if (user.tags.includes(VRCRanks.Veteran)) {
+            return VRCRanks.Veteran;
+        } else if (tags.includes(VRCRanks.Trusted)) {
+            return VRCRanks.Trusted;
+        } else if (tags.includes(VRCRanks.Known)) {
+            return VRCRanks.Known;
+        } else if (tags.includes(VRCRanks.User)) {
+            return VRCRanks.User;
+        } else if (tags.includes(VRCRanks.New)) {
+            return VRCRanks.New;
+        } else if (!tags.includes(VRCRanks.Veteran) &&
+            !tags.includes(VRCRanks.Trusted) &&
+            !tags.includes(VRCRanks.Known) &&
+            !tags.includes(VRCRanks.User) &&
+            !tags.includes(VRCRanks.New) &&
+            !tags.includes(VRCRanks.Nuisance)) {
+
+            return VRCRanks.Visitor;
+        } else if (tags.includes(VRCRanks.Nuisance)) {
+            return VRCRanks.Nuisance;
+        } else {
+            return undefined;
+        }
+
+    }
+
+
+    /**
+     * Get the rank of the user as a string.
+     * @param user The User or Current User object to get the rank name from.
+     * @returns The rank name of the user. If the user has no rank tag, returns "Rank not found."
+     */
+    export function getVRCRankName(user: VRCAPI.Users.Models.User | VRCAPI.Users.Models.CurrentUser): string {
+        const rank = getVRCRankTag(user);
+        if (!rank) {
+            return "Rank not found.";
+        }
+        return VRCRanksName[rank];
+    }
+
+    /**
+     * Get if User is a VRC+ subscriber.
+     * @param user The User object to check if it is a VRChat User with a current VRC+ subscription.
+     * @returns `true` if the user is a VRChat User with a current VRC+ subscription, `false` otherwise.
+     */
+    export function isVRCPlusSubcriber(user: VRCAPI.Users.Models.User): boolean {
+        return user.tags.includes("system_supporter");
+
+    }
+
+    /**
+     * Get if User has a troll tag.
+     * @param user The User object to check if it is a VRChat User with a troll tag.
+     * @returns `true` if the user is a VRChat User with a troll tag, `false` otherwise.
+     */
+    export function isUserTroll(user: VRCAPI.Users.Models.User): boolean {
+        return user.tags.includes(VRCRanks.Nuisance) || user.tags.includes("system_probable_troll");
+    }
```

Comment:
This is alright, i don't know much about how VRC works, but try not to be opnionated on the way you do it, maybe a Nuisance rank player is not a troll, just a child etc, you could have the opnion most of them are probably, but if the system has a tag for that you should just use what the system has.

Reply:
Remember other people will use the Wrapper, they may have different opnions than you.

Reply:
This is completely irelevant, indeed that is how vrchat tags works, there is indeed a tag called nuisance and a tag called troll, for both cases I simply made a function to know if a user has any of the tags, but normally, if a user has one of them, they have the other one, I just played it safe.